examples/common.cpp

```diff
@@ -34,9 +38,25 @@ bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
     // std::thread::hardware_concurrency may not be equal to the number of cores, or may return 0.
 #ifdef __linux__
     std::ifstream cpuinfo("/proc/cpuinfo");
-    params.n_threads = std::count(std::istream_iterator<std::string>(cpuinfo),
-                                  std::istream_iterator<std::string>(),
-                                  std::string("processor"));
+    std::string line;
+    while (std::getline(cpuinfo, line)) {
+        if (line.find("cpu cores") != std::string::npos) {
+            line.erase(0, line.find(": ") + 2);
+            params.n_threads = std::stoul(line);
```

Comment:
Perhaps add a try-catch block here in case stoul does not parse an integer.

```
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoul
Aborted (core dumped)
```